### ISSUE TITLE
[no-use-before-define] Typescript Optional Chaining False Positive

### ISSUE THREAD

**Issue by Skillz4Killz** [typescript-eslint/typescript-eslint#1182]
**Repro**

```JSON
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2019,
    "sourceType": "module"
  },
  "extends": ["plugin:@typescript-eslint/recommended", "prettier/@typescript-eslint", "plugin:prettier/recommended"],
  "rules": {
    "@typescript-eslint/explicit-function-return-type": "off"
  }
}
```

```TS
const language = guildSettings?.language || `en-US`
```

**Expected Result**
This should not throw an error as it is valid syntax.

**Actual Result**
It throws a false positive error thinking that the `language` variable is being used but its actually a property call using the new Typescript optional chaining in 3.7 version.

**Additional Info**

```shell
46:50  error  'language' was used before it was defined  @typescript-eslint/no-use-before-define

✖ 1 problem (1 error, 0 warnings)
```
**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.6.1` |
| `@typescript-eslint/parser`        | `2.6.1` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.5.1` |
| `node`                             | `12.13.0` |
| `npm`                              | `6.12.0` |

**Comment by bradzacher** [comment]
Duplicate of #1104

Please use the search
